pkg/runtime/pubsub/outbox.go

```diff
@@ -184,6 +185,10 @@ func (o *outboxImpl) PublishInternal(ctx context.Context, stateStore string, ope
 				if sr.ContentType != nil {
 					contentType = *sr.ContentType
 				}
+
+				if ct, ok := sr.Metadata[metadata.ContentType]; ok {
```

Comment:
This change presents a problem because it overrides the state transaction level content type that exists today, on line 185.
The explicit field is better to take priority. It makes more sense to only apply the metadata content type if the state transaction level content type is empty